### PR TITLE
ABC-93 Support for loading emojis asynchronously in posts

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
Support for loading emojis asynchronously in posts.

#### Ticket Link
https://mattermost.atlassian.net/browse/ABC-93

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)
- [x] Has redux changes (mattermost/mattermost-redux#379)
- [x] Has UI changes

### CODE REVIEW

**Comment by jwilander** [comment]
@jason I loaded some custom emojis onto this server (note they won't work with autocomplete or the picker because those PRs are pending)

File: tests/utils/emoticons.test.jsx [hunk 1]
```diff
@@ -27,10 +27,5 @@ describe('Emoticons', () => {
             expect(Emoticons.handleEmoticons('asdf:goat:asdf:dash:asdf', new Map(), emojis)).
                 toEqual('asdf$MM_EMOTICON0asdf$MM_EMOTICON1asdf');
         });
```
Comment (saturninoabril) [inline tests/utils/emoticons.test.jsx]: Should remove third argument (`emojis`) in the above tests

File: tests/utils/emoticons.test.jsx [hunk 2]
```diff
@@ -27,10 +27,5 @@ describe('Emoticons', () => {
...
-
-        test('shouldn\'t replace invalid emoticons', () => {
-            expect(Emoticons.handleEmoticons(':asdf: :goat : : dash:', new Map(), emojis)).
-                toEqual(':asdf: :goat : : dash:');
-        });
     });
```
Comment (saturninoabril) [inline tests/utils/emoticons.test.jsx]: Maybe just update the test like:
```
test('shouldn\'t replace invalid emoticons', () => {		
    expect(Emoticons.handleEmoticons(':goat : : dash:', new Map())).		
        toEqual(':goat : : dash:');		
});
```

File: tests/utils/emoticons.test.jsx (hunk 1 above)
Comment (jwilander) [reply tests/utils/emoticons.test.jsx]: Ah, nice catch

**Comment by jwilander** [comment]
@saturninoabril updated

**Pull request merged**